test/Sync.t.sol

```diff
@@ -0,0 +1,248 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import {Test} from "forge-std/Test.sol";
+import {GasSnapshot} from "forge-gas-snapshot/GasSnapshot.sol";
+import {MockERC20} from "solmate/test/utils/mocks/MockERC20.sol";
+
+import {Deployers} from "./utils/Deployers.sol";
+import {IHooks} from "../src/interfaces/IHooks.sol";
+import {Currency, CurrencyLibrary} from "../src/types/Currency.sol";
+import {IPoolManager} from "../src/interfaces/IPoolManager.sol";
+import {PoolSwapTest} from "../src/test/PoolSwapTest.sol";
+import {IUnlockCallback} from "../src/interfaces/callback/IUnlockCallback.sol";
+import {PoolKey} from "../src/types/PoolKey.sol";
+import {BadRouter} from "../src/test/BadRouter.sol";
+import {ActionsRouter, Actions} from "../src/test/ActionsRouter.sol";
+import {SafeCast} from "../src/libraries/SafeCast.sol";
+import {Reserves} from "../src/libraries/Reserves.sol";
+
+contract SyncTest is Test, Deployers, GasSnapshot {
+    using CurrencyLibrary for Currency;
+
+    // PoolManager has no balance of currency2.
+    Currency currency2;
+    BadRouter badRouter;
+    ActionsRouter router;
+
+    function setUp() public {
+        initializeManagerRoutersAndPoolsWithLiq(IHooks(address(0)));
+        currency2 = deployMintAndApproveCurrency();
+        badRouter = new BadRouter(manager);
+        router = new ActionsRouter(manager);
+    }
+
+    function test_sync_balanceIsZero() public {
+        assertEq(currency2.balanceOf(address(manager)), uint256(0));
+        uint256 balance = manager.sync(currency2);
+
+        assertEq(uint256(balance), 0);
+        assertEq(manager.getReserves(currency2), 0);
+    }
+
+    function test_sync_balanceIsNonZero() public {
+        uint256 currency0Balance = currency0.balanceOf(address(manager));
+        assertGt(currency0Balance, uint256(0));
+
+        // Without calling sync, getReserves should revert.
+        vm.expectRevert(Reserves.ReservesMustBeSynced.selector);
+        manager.getReserves(currency0);
+
+        uint256 balance = manager.sync(currency0);
+        assertEq(balance, currency0Balance, "balance not equal");
```

Comment:
nice to add another 
``` 
       assertEq(manager.getReserves(currency2), balance);
```

to show now that sync has been called it will not revert